source/extensions/filters/common/auth/attribute_context.cc

```diff
@@ -0,0 +1,91 @@
+#include "extensions/filters/common/auth/attribute_context.h"
+
+#include "common/http/utility.h"
+#include "common/network/utility.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Filters {
+namespace Common {
+namespace Auth {
+
+void AttributeContextUtils::setSourcePeer(envoy::service::auth::v2alpha::AttributeContext& context,
+                                          const Network::Connection& connection,
+                                          const std::string& service) {
+  setPeer(*context.mutable_source(), connection, service, false);
+}
+
+void AttributeContextUtils::setDestinationPeer(
+    envoy::service::auth::v2alpha::AttributeContext& context,
+    const Network::Connection& connection) {
+  setPeer(*context.mutable_destination(), connection, "", true);
+}
+
+void AttributeContextUtils::setPeer(envoy::service::auth::v2alpha::AttributeContext_Peer& peer,
+                                    const Network::Connection& connection,
+                                    const std::string& service, const bool local) {
+
+  auto addr = local ? connection.localAddress() : connection.remoteAddress();
+  Envoy::Network::Utility::addressToProtobufAddress(*addr, *peer.mutable_address());
+
+  if (!service.empty()) {
+    peer.set_service(service);
+  }
+
+  auto ssl = const_cast<Ssl::Connection*>(connection.ssl());
+  if (ssl) {
+    std::string principal = local ? ssl->uriSanLocalCertificate() : ssl->uriSanPeerCertificate();
+    if (principal.empty()) {
+      principal = local ? ssl->subjectLocalCertificate() : ssl->subjectPeerCertificate();
+    }
+    peer.set_principal(principal);
+  }
+}
+
+void AttributeContextUtils::setHttpRequest(
+    envoy::service::auth::v2alpha::AttributeContext& context,
+    const Envoy::Http::StreamDecoderFilterCallbacks* callbacks,
+    const Envoy::Http::HeaderMap& headers) {
+
+  auto req = context.mutable_request()->mutable_http();
+  auto sdfc = const_cast<Envoy::Http::StreamDecoderFilterCallbacks*>(callbacks);
+
+  auto start = ProtobufUtil::TimeUtil::MicrosecondsToTimestamp(
+      sdfc->requestInfo().startTime().time_since_epoch().count());
+  context.mutable_request()->mutable_time()->MergeFrom(start);
+
+  auto proto = sdfc->requestInfo().protocol();
+  if (proto) {
+    req->set_protocol(Envoy::Http::Utility::getProtocolString(proto.value()));
+  }
+
+  req->set_id(std::to_string(sdfc->streamId()));
+  req->set_size(sdfc->requestInfo().bytesReceived());
```

Comment:
Is this capturing the entire request size or only what has been received at the time of the call?

Reply:
It will be at the time of the call. If ext_authz and RBAC are applied in `decodeHeaders`, this field isn't exactly helpful. Just porting the existing functionality. Perhaps it'd be best to set this to -1 for now per the docs on the proto message?

Reply:
Or look for the content-length header?

Reply:
I'm not sure. This could be a setting in the RBAC filter to buffer requests, although I'm not sure why we would want to use request size in RBAC at all.

Reply:
I can't think of a use case that RBAC uses request.size. 

Maybe use "content-length header" makes more sense. The request.size value should not depend on where the filter is placed.

Reply:
works for me!